### PR TITLE
Add husky config

### PR DESCRIPTION
**Pull request by pankominek**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Setup pre-commit hook using husky. It will run our maintenance tools before committing changes.

### Requirements
1. When doing commit, Husky pre-commit hook should run `yarn lint`, and `yarn pretty` scripts
2. Husky should run pre-commit check for commit messages. We should use [this library](https://github.com/conventional-changelog/commitlint)

### Review

#### .husky/pre-commit
The pre-commit hook is set up correctly to run `yarn test` and `yarn lint`. However, it does not meet the task requirements as it does not run `yarn pretty`. Also, there is no check for commit messages as required.

```diff
#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

+yarn pretty
yarn test
yarn lint
```

#### package.json
The husky package is added correctly to the devDependencies. The `prepare` script is also added correctly to install husky. However, the commitlint library is not added to the devDependencies and there is no script to check commit messages.

```diff
"devDependencies": {
+    "@commitlint/cli": "^13.2.1",
+    "@commitlint/config-conventional": "^13.2.0",
    "eslint": "^8.49.0",
    "eslint-plugin-react": "latest",
    "eslint-plugin-storybook": "latest",
    "husky": "^8.0.3",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^27.0.0",
    "preact-cli": "^3.4.2",
}

"scripts": {
    "clean": "rimraf dist/",
    "copy-files": "copyfiles -u 1 src/**/*.css dist/",
    "build": "yarn clean && tsc -b && yarn copy-files",
    "lint": "eslint src",
+   "commitmsg": "commitlint -E HUSKY_GIT_PARAMS",
    "prepare": "husky install"
}
```

In conclusion, the code changes do not fully cover the task requirements. The missing parts are the `yarn pretty` script in the pre-commit hook and the commit message check using commitlint.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review

Overall, the code changes cover the task requirements effectively. The changes include setting up a pre-commit hook using Husky, running maintenance tools before committing changes, and implementing pre-commit checks for commit messages using the commitlint library. The code changes are consistent with the task requirements and follow best practices.

Here are some specific observations:

### .babelrc
- The changes in the `.babelrc` file correctly update the configuration for the `@babel/preset-react` plugin.

### .husky/pre-commit
- The added `pre-commit` script in the `.husky/pre-commit` file runs the `yarn pretty`, `yarn test`, and `yarn lint` scripts before committing changes. This fulfills the requirement of running maintenance tools before committing.

### .storybook/preview.js
- The changes in the `.storybook/preview.js` file correctly update the export statement.

### LICENSE.md
- The changes in the `LICENSE.md` file do not affect the content or requirements of the license. They only remove unnecessary formatting and whitespace changes.

### package.json
- The added `commitmsg` script in the `package.json` file runs the `commitlint` command with the appropriate parameters to check commit messages against conventional commit standards. This fulfills the requirement of running pre-commit checks for commit messages.

### src/Introduction.stories.mdx
- The changes in the `src/Introduction.stories.mdx` file correctly update the import statements for the SVG assets.

### src/inputs/Button/Button.module.css
- The changes in the `src/inputs/Button/Button.module.css` file correctly update the CSS class names and styles.

### src/inputs/Button/readme.md
- The changes in the `src/inputs/Button/readme.md` file correctly update the formatting and wording of the requirements.

### src/inputs/Checkbox/readme.md
- The changes in the `src/inputs/Checkbox/readme.md` file correctly update the formatting and wording of the requirements.

### src/inputs/Switch/readme.md
- The changes in the `src/inputs/Switch/readme.md` file correctly update the formatting and wording of the requirements.

### src/inputs/TextField/readme.md
- The changes in the `src/inputs/TextField/readme.md` file correctly update the formatting and wording of the requirements.

### src/inputs/readme.md
- The changes in the `src/inputs/readme.md` file correctly update the formatting and wording of the content.

### src/navigation/Link/readme.md
- The changes in the `src/navigation/Link/readme.md` file correctly update the formatting and wording of the requirements.

### src/navigation/readme.md
- The changes in the `src/navigation/readme.md` file correctly update the formatting and wording of the content.

### tsconfig.json
- The changes in the `tsconfig.json` file correctly update the `exclude` property to exclude the `dist` and `node_modules` directories.

Overall, the code changes effectively implement the required functionality and meet the task requirements. Well done!
